woocommerce/payment-gateway/class-sv-wc-payment-gateway-payment-form.php

```diff
@@ -99,7 +99,22 @@ protected function add_hooks() {
 		add_action( "wc_{$gateway_id}_payment_form_end",   array( $this, 'render_fieldset_end' ), 5 );
 
 		// payment form JS
-		add_action( "wc_{$gateway_id}_payment_form_end",   array( $this, 'render_js' ), 5 );
+		add_action( 'wp_footer', [ $this, 'load_render_js' ] );
+	}
+
+
+	/**
+	 * Loads payment gateway render script if checkout page or pay page.
+	 *
+	 * @since 5.10.5
+	 */
+	public function load_render_js() {
+
+		// also make sure it's not order received page
+		if ( ! is_order_received_page() && ( is_checkout() || is_checkout_pay_page() ) ) {
+			// payment form JS
+			add_action( 'wp_footer', [ $this, 'render_js' ] );
```

Comment:
This is being called when already inside the `wp_footer` action, so it doesn't end up rendering the JS.

Rather than the `load_render_js` wrapping function, we should just use `add_action( 'wp_footer', [ $this, 'render_js' ] );` in `add_hooks()`, then in `render_js` add the logic to bail early if not on the proper pages

Reply:
Because `render_js` method gets overridden in multiple gateways, so I thought of a more generic way of handling rendering the JS without the need for the gateways to update their code.

Reply:
Yeah that's a good point - it does make it easier for gateways to upgrade.

But the main issue still remains and I think is the source of the problem here. `load_render_js` is hooked to `wp_footer`, so when that method fires it will already be after `wp_footer` has been fired. But then it's trying to hook into `wp_footer` again to call `render_js`, but that won't fire again.

It seems to me like this could be simplified and fixed by just calling `render_js()` directly inside `load_render_js()` (though I'd recommend calling it `maybe_render_js()` instead, to keep with what we've done elsewhere). No need for the double `add_action( 'wp_footer'`.

Reply:
@ChaseWiseman after more digging and testing based on our conversation today, looks like each gateway plugin handles the issue differently, ex: after loading the `render_js` in footer, Bambora throws a ton of console errors but still works! I think these errors can be handled without a problem, but the issue here I believe it comes to each gateway implementation, I don't think there is a solid solution for the FW that fits the majority of the gateways.


https://user-images.githubusercontent.com/3770036/111697121-cede7480-883d-11eb-98c2-24abc340dae8.mp4